File: fixtures/scrapy_output/options.py
from scrapy import Request, Spider


class MySpider(Spider):
    name = 'myspider'

    def start_requests(self):
        headers = {
            'Pragma': 'no-cache',
            'Access-Control-Request-Method': 'POST',
            'Origin': 'https://alexa.amazon.de',
            'Accept-Encoding': 'gzip, deflate, br',
            'Accept-Language': 'de-DE,de;q=0.8,en-US;q=0.6,en;q=0.4',
            'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/61.0.3163.100 Safari/537.36',
            'Accept': '*/*',
            'Cache-Control': 'no-cache',
            'Referer': 'https://alexa.amazon.de/spa/index.html',
            'Connection': 'keep-alive',
            'DNT': '1',
            'Access-Control-Request-Headers': 'content-type,csrf',
        }
        yield Request(
            'https://layla.amazon.de/api/tunein/queue-and-play?deviceSerialNumber=xxx^&deviceType=xxx^&guideId=s56876^&contentType=station^&callSign=^&mediaOwnerCustomerId=xxx',
            callback=self.parse,
            method='OPTIONS',
            headers=headers,
        )

    def parse(self, response):
        pass

File: test/fixtures/java-jsoup/options.java
import java.io.File;
import java.io.FileInputStream;
import java.io.IOException;
import org.jsoup.Connection;
import org.jsoup.Jsoup;


class Main {

	public static void main(String[] args) throws IOException {
		Connection.Response response = Jsoup.connect("http://localhost:28139/api/tunein/queue-and-play?deviceSerialNumber=xxx&deviceType=xxx&guideId=s56876&contentType=station&callSign=&mediaOwnerCustomerId=xxx")
			.header("Pragma", "no-cache")
			.header("Access-Control-Request-Method", "POST")
			.header("Origin", "https://alexa.amazon.de")
			.header("Accept-Encoding", "gzip, deflate, br")
			.header("Accept-Language", "de-DE,de;q=0.8,en-US;q=0.6,en;q=0.4")
			.userAgent("Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/61.0.3163.100 Safari/537.36")
			.header("Accept", "*/*")
			.header("Cache-Control", "no-cache")
			.header("Referer", "https://alexa.amazon.de/spa/index.html")
			.header("Connection", "keep-alive")
			.header("DNT", "1")
			.header("Access-Control-Request-Headers", "content-type,csrf")
			.method(org.jsoup.Connection.Method.OPTIONS)
			.ignoreContentType(true)
			.execute();

		System.out.println(response.parse());
	}
}
